### ISSUE TITLE
Low priority: docs won't build on Linux Mint 17

### ISSUE THREAD

**Issue by Seanny123** [nengo/nengo#477]
Here are the contents of the error log. I'm running Linux Mint 17 with iPython 2.2.0. With the exact same configuration, it runs on Ubuntu 14.04, so I really don't know what's going on here and it's not that important, but I thought I would note it down just in case it comes up again.

    # Sphinx version: 1.2.3
    # Python version: 2.7.6
    # Docutils version: 0.12 release
    # Jinja2 version: 2.7.2
    # Loaded extensions:
    #   sphinx.ext.autodoc from /usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/ext/autodoc.pyc
    #   sphinx.ext.mathjax from /usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/ext/mathjax.pyc
    #   numpydoc from /home/saubin/git/nengo-docs/src/numpydoc/numpydoc/__init__.py
    #   sphinx.ext.viewcode from /usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/ext/viewcode.pyc
    #   nengo.utils.docutils from /home/saubin/git/nengo-docs/nengo/utils/docutils.pyc
    #   sphinx.ext.todo from /usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/ext/todo.pyc
    #   sphinx.ext.autosummary from /usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/ext/autosummary/__init__.pyc
    #   sphinx.ext.oldcmarkup from /usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/ext/oldcmarkup.pyc
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/cmdline.py", line 254, in main
        app.build(force_all, filenames)
      File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/application.py", line 220, in build
        self.builder.build_update()
      File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/builders/__init__.py", line 214, in build_update
        'out of date' % len(to_build))
      File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/builders/__init__.py", line 234, in build
        purple, length):
      File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/builders/__init__.py", line 134, in status_iterator
        for item in iterable:
      File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/environment.py", line 478, in update_generator
        self.read_doc(docname, app=app)
      File "/usr/local/lib/python2.7/dist-packages/Sphinx-1.2.3-py2.7.egg/sphinx/environment.py", line 628, in read_doc
        pub.publish()
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/core.py", line 217, in publish
        self.settings)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/readers/__init__.py", line 72, in read
        self.parse()
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/readers/__init__.py", line 78, in parse
        self.parser.parse(self.input, document)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/__init__.py", line 172, in parse
        self.statemachine.run(inputlines, document, inliner=self.inliner)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 170, in run
        input_source=document['source'])
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/statemachine.py", line 239, in run
        context, state, transitions)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/statemachine.py", line 460, in check_line
        return method(match, context, next_state)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 2961, in text
        self.section(title.lstrip(), source, style, lineno + 1, messages)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 327, in section
        self.new_subsection(title, lineno, messages)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 395, in new_subsection
        node=section_node, match_titles=True)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 282, in nested_parse
        node=node, match_titles=match_titles)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 195, in run
        results = StateMachineWS.run(self, input_lines, input_offset)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/statemachine.py", line 239, in run
        context, state, transitions)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/statemachine.py", line 460, in check_line
        return method(match, context, next_state)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 2299, in explicit_markup
        nodelist, blank_finish = self.explicit_construct(match)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 2311, in explicit_construct
        return method(self, expmatch)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 2054, in directive
        directive_class, match, type_name, option_presets)
      File "/usr/local/lib/python2.7/dist-packages/docutils-0.12-py2.7.egg/docutils/parsers/rst/states.py", line 2103, in run_directive
        result = directive_instance.run()
      File "/home/saubin/git/nengo-docs/nengo/utils/docutils.py", line 97, in run
        evaluated_html = ipext.export_html(nb_eval)
      File "/home/saubin/git/nengo-docs/nengo/utils/ipython.py", line 141, in export_html
        output, resources = exporter.from_notebook_node(nb)
      File "/usr/local/lib/python2.7/dist-packages/IPython/nbconvert/exporters/templateexporter.py", line 209, in from_notebook_node
        nb_copy, resources = super(TemplateExporter, self).from_notebook_node(nb, resources, **kw)
      File "/usr/local/lib/python2.7/dist-packages/IPython/nbconvert/exporters/exporter.py", line 119, in from_notebook_node
        if 'language' in nb['metadata']:
    TypeError: string indices must be integers, not str

**Comment by jgosmann** [comment]
Having a similar error on OS X Yosemite:
```
(nengodev)v1020-wn-137-238:docs|docs> python `which sphinx-build` . build
Running Sphinx v1.2.3
loading pickled environment... not yet created
No builder selected, using default: html
building [html]: targets for 56 source files that are out of date
updating environment: 56 added, 0 changed, 0 removed
reading sources... [ 21%] examples/basal_ganglia
Exception occurred:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/IPython/nbconvert/exporters/exporter.py", line 119, in from_notebook_node
    if 'language' in nb['metadata']:
TypeError: string indices must be integers, not str
The full traceback has been saved in /var/folders/t7/99gtw6151wz7r76dzhxl08vr0000gn/T/sphinx-err-hcrQoP.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://bitbucket.org/birkenfeld/sphinx/issues/>. Thanks!
```

**Comment by tbekolay** [comment]
Apparently this occurred whenever a notebook raised an exception; it returned the exception as a string, which wasn't super helpful (but it was what the library I took some of the code from did). Now (or at least, once #371 is merged) it will give a warning but keep on going.
